### DISCUSSION TITLE
Search behavior after update

### DISCUSSION THREAD

**Discussion by eliasfank** [codingjoe/django-select2#35] [Ideas]
**Describe the bug**

Obs: I'm not sure between bug or new feature

After #26, the search behavior has changed from AND to OR on splited terms. In my opinion, this behavior is incorrect because when I add two words to the search I would like both to be in the result query. I’ve been using __icontains on the widgets until then to achieve the goal of bringing results that include all the searched words, but now it is not working the same way.

But I also agree that there may be situations where OR is preferred. Is there a possibility to implement the choice between the two forms(AND | OR) of quering? Or how could I do to have the previous AND behavior ([here](https://github.com/codingjoe/django-select2/blob/master/django_select2/forms.py#L410)) with the newest version of the app?

**Comment by codingjoe** [comment]
Hi @eliasfank,

Thanks for reaching out. I agree that this change could be been communicated better. However, I believe this to be the better default. That being said, I am known to be wrong in the past. My suggestion here would be to overwrite the method to your actual needs. `icontains` queries aren't really a good way to search anyway. I would always suggest going with a proper full-test search.

I know this might not be the answer you are looking for, but we will never find a solution that works for everyone. So encouraging people to overwrite `filter_queryset` seems the right thing to do.

Best,
Joe

**Reply by eliasfank** [reply]
Hi @codingjoe, thanks for answering!

Overwriting `filter_queryset` seems to be a good idea. Your suggestion of go with a full search is also appropriate, but I have some peculiarities on my project that makes the search lookup not work. Maybe you know a solution to help me use the search on the widget.

The project is running with postgres and because of the Brazilian portuguese language, using `unaccent` is a must. I've tried tu use both `unaccent` and `search` for querying, but it doesn't works together and raises some errors.

```python
Customer.objects.filter(name__unaccent__icontains)  # this achieved the goal with the select2<=7.5
```
```python
Customer.objects.filter(name__unaccent__search)  # this only works removing the unaccent lookup
```

Do you think it is possible to use full search and unaccent together?

**Reply by codingjoe** [reply]
What database are you using? If you are using PostgreSQL I used proper [full-text search](https://docs.djangoproject.com/en/stable/ref/contrib/postgres/search/). It's vastly more powerful using language based stemming.
